schema/field.go

```diff
@@ -113,6 +116,14 @@ func (f *Field) UnmarshalJSON(data []byte) error {
 		return err
 	}
 	*f = Field(*u)
+
+	if (*f).Constraints.Pattern != "" {
```

Comment:
I believe there is no need for dereferencing here and bellow within this method. Noticed I renamed `compiledRegexp` to `compiledPattern` to make it easy to associate to the Patter field. Suggestion:

```go
if f.Constraints.Pattern != "" {
   p, err := regexp.Compile(f.Constraints.Pattern) 
   if err != nil {
      return err
   }
   f.Constraints.compiledPattern = p
}
```